src/client/renderer/r_program.c

```diff
@@ -481,15 +488,13 @@ static void R_LoadShader(GLenum type, const char *name, r_shader_t *out_shader)
 	glCompileShader(out_shader->id);
 
 	glGetShaderiv(out_shader->id, GL_COMPILE_STATUS, &e);
+
 	if (!e) {
 		glGetShaderInfoLog(out_shader->id, sizeof(log) - 1, NULL, log);
 		Com_Warn("%s: %s\n", out_shader->name, log);
 
 		glDeleteShader(out_shader->id);
 		memset(out_shader, 0, sizeof(*out_shader));
-
-		Fs_Free(buf);
-		return;
```

Comment:
Used to have to but now that the functions are split, no. The code after it will still work if a shader fails to load (the program will fail to compile which is what we want).